File: src/xunit.installer.templates/MVC/CSharp/Common/TestDoubles/StubRequest.cs
using System.Web;

namespace $safeprojectname$
{
    public class StubRequest : HttpRequestBase
    {
        string relativeUrl;

        public StubRequest(string relativeUrl)
        {
            this.relativeUrl = relativeUrl;
        }

        public override string AppRelativeCurrentExecutionFilePath
        {
            get { return relativeUrl; }
        }

        public override string PathInfo
        {
            get { return ""; }
        }
    }
}

File: src/xunit.installer.templates/MVC/VisualBasic/Common/TestDoubles/StubRequest.vb
Imports System.Web

Public Class StubRequest
    Inherits HttpRequestBase

    Private relativeUrl As String

    Public Sub New(ByVal relativeUrl As String)
        Me.relativeUrl = relativeUrl
    End Sub

    Public Overrides ReadOnly Property AppRelativeCurrentExecutionFilePath() As String
        Get
            Return relativeUrl
        End Get
    End Property

    Public Overrides ReadOnly Property PathInfo() As String
        Get
            Return ""
        End Get
    End Property
End Class